### PR TITLE
Add origin experiments check for amp-list

### PR DESCRIPTION
**Pull request by cathyxz**
Adds a check to see if `amp-list-load-more` is within the set of origin experiments enabled for this given page. Some refactoring happened due to the fact that this check returns a promise and not a direct boolean. I debated adding a piece of code that would automatically use toggleExperiment to turn on all experiments in origin trial during runtime, but decided against this as it could inadvertently enabled experiments for all users who had once visited an origin trial page. I think a more ideal solution would be to implement a check for `isOriginExperimentOn` per @choumx 's original design, but I'm not sure that was ever implemented or where it should go (given the current async loading of `origin-experiments-impl.js` etc).

### CODE REVIEW

**Comment by choumx** [comment]
Drive by comment on a WIP PR, but while we're in a refactor-y mood maybe we should split up amp-list which is 1K+ LOC already. 😄

**Comment by cathyxz** [comment]
I've moved about 300 LOC related to the load-more feature out of `amp-list.js`. But the reason it is so long is that it has a few structural problems (given what we've extended it to be used for), and many inconvenient assumptions that we cannot break due to backwards compat. The upcoming V2 is where we're hoping to address that.

File: extensions/amp-list/0.1/amp-list.js [hunk 1]
```diff
@@ -156,6 +149,17 @@ export class AmpList extends AMP.BaseElement {
     this.ssrTemplateHelper_ = new SsrTemplateHelper(
         TAG, viewer, this.templates_);
 
+    const originTrialPromise = originExperimentsForDoc(this.getAmpDoc())
```
Comment (aghassemi) [inline extensions/amp-list/0.1/amp-list.js]: discussed offline but we want short-circuit and `promise.resolve` immediately if no `load-more`

File: extensions/amp-list/0.1/service/load-more-service.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
+/**
```
Comment (aghassemi) [inline extensions/amp-list/0.1/service/load-more-service.js]: nice refactoring

**Pull request merged**